docs/design/pattern_matching.md

```diff
@@ -169,46 +191,27 @@ fn F() -> i32 {
 }
 ```
 
-When a new object needs to be created for the binding, the lifetime of the bound
-value matches the scope of the binding.
-
-```carbon
-class NoisyDestructor {
-  fn Make() -> Self { return {}; }
-  impl i32 as ImplicitAs(NoisyDestructor) {
-    fn Convert[me: i32]() -> Self { return Make(); }
-  }
-  destructor {
-    Print("Destroyed!");
-  }
-}
-
-fn G() {
-  // Does not print "Destroyed!".
-  let n: NoisyDestructor = NoisyDestructor.Make();
-  Print("Body of G");
-  // Prints "Destroyed!" here.
-}
-
-fn H(n: i32) {
-  // Does not print "Destroyed!".
-  let (v: NoisyDestructor, w: i32) = (n, n);
-  Print("Body of H");
-  // Prints "Destroyed!" here.
-}
-```
+`self` can be used instead of an identifier only if the pattern is an implicit
+parameter of a member function (optionally enclosed in a `var` pattern). This
```

Comment:
Is this "of a member function" a new requirement? That seems like a separate change to this proposal, and it's a rule that there's been some uncertainty about, so procedurally I think we shouldn't change the status of that question as a drive-by here and instead leave it to a separate proposal. (If this wording is just being moved or copied from elsewhere and we're not changing its status, that's fine.)